core/src/test/java/org/pentaho/di/connections/ConnectionManagerTest.java

```diff
@@ -255,12 +258,30 @@ public void testDefaultPropertiesNotNull() {
     Assert.assertNotNull( connectionDetails.getProperties() );
     Assert.assertNotNull( connectionDetails.getProperties().get( "baRoles" ) );
   }
+  @Test
+  public void testConnection() throws KettleException {
+    addVfsProvider();
+    TestConnectionDetails testConnectionDetails = new TestConnectionDetails();
+    connectionManager.test( testConnectionDetails, new VFSConnectionTestOptions() );
+  }
+
+  @Test
```

Comment:
There still aren't any tests that accomplish my ask - https://github.com/pentaho/pentaho-kettle/pull/9305#discussion_r1595841173
 mainly there isn't a `Mockito.verify` or Using Mockito's agreement captor.

Reply:
Addressed the changes. Added the test cases with Mockito.verify  which verify tests with nonnull/null options.